From 11d217d172a82a072733852a71a69b63bb9f1f59 Mon Sep 17 00:00:00 2001
From: Marcel Siegert <mws@directfb.org>
Date: Thu, 11 Jan 2007 18:08:22 +0000
Subject: [PATCH] fix mentioned compile errors for kernel 2.6.20 i took 2.6.20 as the minimum kernel version for the additional includes maybe that appeared already at 2.6.19

linux-fusion/linux/drivers/char/fusion/fusiondev.c: In function 'fusiondev_deinit':
linux-fusion/linux/drivers/char/fusion/fusiondev.c:253: warning: implicit declaration of function 'ClearPageReserved'
linux-fusion/linux/drivers/char/fusion/fusiondev.c:253: error: invalid use of undefined type 'struct page'
linux-fusion/linux/drivers/char/fusion/fusiondev.c: In function 'fusion_mmap':
linux-fusion/linux/drivers/char/fusion/fusiondev.c:1011: error: dereferencing pointer to incomplete type
---
 drivers/char/fusion/fusiondev.c |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/drivers/char/fusion/fusiondev.c b/drivers/char/fusion/fusiondev.c
index 2da2d68..75e99ea 100644
--- a/drivers/char/fusion/fusiondev.c
+++ b/drivers/char/fusion/fusiondev.c
@@ -24,6 +24,11 @@
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 18)
 #include <linux/devfs_fs_kernel.h>
 #endif
+#if LINUX_VERSION_CODE > KERNEL_VERSION(2, 6, 19)
+#include <linux/page-flags.h>
+#include <linux/mm.h>
+#endif
+
 #include <linux/proc_fs.h>
 #include <linux/poll.h>
 #include <linux/init.h>
-- 
1.7.3.3

